src/Evolution/Systems/GrMhd/GhValenciaDivClean/StressEnergy.hpp

```diff
@@ -0,0 +1,98 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace grmhd::GhValenciaDivClean {
+
+/*!
+ * \brief Add in the stress-energy source term to the \f$\Pi\f$ evolved variable
+ * of the Generalized Harmonic system.
+ *
+ * \details The only stress energy source term in the Generalized Harmonic
+ * evolution equations is in the equation for \f$\Pi_{a b}\f$:
+ *
+ * \f[
+ * \partial_t \Pi_{ab} + \text{(spatial derivative terms)} =
+ * \text{(GH source terms)}
+ * - 16 \pi \alpha (T_{ab} - \frac{1}{2} g_{a b} T^c{}_c)
+ * \f]
+ *
+ * This function adds that contribution to the existing value of `dt_pi`. The
+ * spacetime terms in the GH equation should be computed before passing the
+ * `dt_pi` to this function for updating.
+ *
+ * \see `GeneralizedHarmonic::TimeDerivative` for details about the spacetime
+ * part of the tiem derivative calculation.
+ */
+void add_stress_energy_term_to_dt_pi(
+    gsl::not_null<tnsr::aa<DataVector, 3>*> dt_pi,
+    gsl::not_null<Scalar<DataVector>*> stress_energy_trace_buffer,
+    const tnsr::aa<DataVector, 3>& stress_energy,
+    const tnsr::AA<DataVector, 3>& inverse_spacetime_metric,
+    const tnsr::aa<DataVector, 3>& spacetime_metric,
+    const Scalar<DataVector>& lapse) noexcept;
+
+/*!
+ * \brief Calculate the stress-energy tensor \f$T_{\mu \nu}\f$ associated with
+ * the matter part of the GRMHD system.
+ *
+ * \details The stress energy tensor is needed to compute the backreaction of
+ * the matter to the spacetime degrees of freedom. The stress energy is
+ * calculated as described in \cite Moesta2013dna :
+ *
+ * \f[
+ * T_{\mu \nu} = \rho h^* u_\mu u_\nu + P^* g_{\mu \nu} - b_\mu b_\nu
+ * \f]
+ *
+ * where \f$u_\mu\f$ is the four-velocity, \f$\rho\f$ is the rest mass density,
+ * \f$h^*\f$ is the magnetically modified enthalpy, and \f$b_{\mu}\f$ is the
+ * comoving magnetic field.
+ *
+ * The spatial components of the four velocity \f$u_\mu\f$ are
+ *
+ * \f[
+ * u_i = W v_i,
+ * \f]
+ *
+ * and the time component is
+ *
+ * \f[
+ * u_0 = - \alpha W + \beta^i u_i.
+ * \f].
+ *
+ * The magnetically modified enthalpy is
+ *
+ * \f[
+ * h^* = 1 + \epsilon + (P + b^2) / \rho.
+ * \f]
+ *
+ * The comoving magnetic field is computed via
+ *
+ * \f{align}{
+ * b_i &= B_i / W + v_i W v^k B_k\\
+ * b_0 &= - \alpha W v^i B_i + \beta^i b_i
+ * \f}
+*/
+void stress_energy(
```

Comment:
is it not easier to just compute the trace-reverse stress energy tensor immediately? That's:
```
rho h^* (u_a u_b + g_{ab} / 2) - p g_{ab} / 2 - b_a b_b
```
if I did the math right. I think if one expands `rho h = rho + rho epsilon + p` then it _might_ get a bit simpler still, but in either case this seems to be less or about the same amount of work as calculating `T_{ab}`, but we don't need to do a generic trace reversal later. 

Thoughts?

Reply:
Yeah, I think that makes the most sense -- the GH source will be the primary use of the stress-energy, so I agree it's easiest to just provide the trace-reversed version from the get-go.